plugins/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/requirement/OpenShift3ProjectRequirement.java

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v 1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributor:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.reddeer.requirement;
+
+import static org.junit.Assert.assertFalse;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.apache.commons.lang.StringUtils;
+import org.jboss.reddeer.common.logging.Logger;
+import org.jboss.reddeer.common.wait.TimePeriod;
+import org.jboss.reddeer.common.wait.WaitUntil;
+import org.jboss.reddeer.junit.requirement.Requirement;
+import org.jboss.tools.openshift.core.connection.Connection;
+import org.jboss.tools.openshift.reddeer.condition.ProjectResourceExists;
+import org.jboss.tools.openshift.reddeer.requirement.OpenShift3ProjectRequirement.OpenShift3Project;
+
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.project.IProjectRequest;
+
+/**
+ * A requirement that makes sure a given project exists. If it doesnt it will get created
+ * 
+ * @author adietish@redhat.com
+ */
+public class OpenShift3ProjectRequirement implements Requirement<OpenShift3Project> {
+
+	private static Logger LOGGER = new Logger(OpenShift3ProjectRequirement.class);
+
+	private OpenShift3Project projectSpec;
+
+	@Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.TYPE)
+    public @interface OpenShift3Project {
+		String connectionURL();
+		String name();
+		String displayName() default StringUtils.EMPTY;
+		String description() default StringUtils.EMPTY;
+	}
+	
+	@Override
+	public boolean canFulfill() {
+		return true;
+	}
+
+	@Override
+	public void fulfill() {
+		assertFalse(StringUtils.isEmpty(projectSpec.name()));
+
+		String projectName = projectSpec.name();
+		Connection connection = OpenShift3ResourceUtils.getConnection(projectSpec.connectionURL());
+		if (OpenShift3ResourceUtils.hasProject(projectName, connection)) {
```

Comment:
You could make negotion of condition and in body execute the code which follows condition.